app/uk/gov/hmrc/tai/viewModels/TaxAccountSummaryViewModel.scala

```diff
@@ -43,63 +42,44 @@ case class TaxAccountSummaryViewModel(header: String,
                                      )
 
 object TaxAccountSummaryViewModel extends ViewModelHelper with TaxAccountFilter {
-  def apply(taxCodeIncomes: Seq[TaxCodeIncome],
-            employments: Seq[Employment],
-            taxAccountSummary: TaxAccountSummary,
+
+  def apply(taxAccountSummary: TaxAccountSummary,
             isAnyFormInProgress: TimeToProcess,
-            nonTaxCodeIncome: NonTaxCodeIncome)(implicit messages: Messages): TaxAccountSummaryViewModel = {
+            nonTaxCodeIncome: NonTaxCodeIncome,
+            livePensionIncomeSources: Seq[IncomeSource],
+            liveEmploymentIncomeSources: Seq[IncomeSource],
+            ceasedEmploymentIncomeSources: Seq[IncomeSource],
+            nonMatchingCeasedEmployments: Seq[Employment])(implicit messages: Messages): TaxAccountSummaryViewModel = {
 
-    val header = messages("tai.incomeTaxSummary.heading.part1") + " " + currentTaxYearRangeHtmlNonBreak
-    val title = messages("tai.incomeTaxSummary.heading.part1") + " " + currentTaxYearRangeHtmlNonBreak
+    val header: String = messages("tai.incomeTaxSummary.heading.part1") + " " + currentTaxYearRangeHtmlNonBreak
+    val title: String = messages("tai.incomeTaxSummary.heading.part1") + " " + currentTaxYearRangeHtmlNonBreak
 
-    val taxFreeAmount = withPoundPrefixAndSign(MoneyPounds(taxAccountSummary.taxFreeAmount, 0))
-    val estimatedIncomeTaxAmount = withPoundPrefixAndSign(MoneyPounds(taxAccountSummary.totalEstimatedTax, 0))
+    val taxFreeAmount: String = withPoundPrefixAndSign(MoneyPounds(taxAccountSummary.taxFreeAmount, 0))
+    val estimatedIncomeTaxAmount: String = withPoundPrefixAndSign(MoneyPounds(taxAccountSummary.totalEstimatedTax, 0))
 
-    val employmentTaxCodeIncomes = taxCodeIncomes filter liveEmployment
-    val employmentViewModels = viewModelsFromMatchingIncomeSources(employmentTaxCodeIncomes, employments)
+    val employmentViewModels: Seq[IncomeSourceViewModel] = liveEmploymentIncomeSources.map(IncomeSourceViewModel(_))
```

Comment:
ViewModels should be as simple as possible, you should inject the IncomeSourceViewModel's in instead of creating them here